tests/main/user-session-env/task.yaml

```diff
@@ -0,0 +1,80 @@
+summary: verify that user environment settings are added
+
+description: |
+    User environment variables are added via /etc/profile.d/snapd.sh (bash/sh
+    specific) or via /lib/environment.d/ helpers. Make sure that at least one of
+    the mechanisms works and XDG_DATA_DIRS and PATH are updated accordingly
+    inside the user session, no matter the shell they use.
+
+# environment generators were introduced in systemd 233, anything before that
+# will not work
+systems:
+   # cannot install zsh
+   - -ubuntu-core-*
+
+environment:
+    TEST_ZSH_USER: test-zsh
+    NO_USER_SESSION_SYSTEMS: ubuntu-14.04 amazon-linux centos-7 debian-9 ubuntu-16.04
+
+prepare: |
+    echo "create a user with a different shell"
+    useradd --create-home --user-group -s /bin/zsh "$TEST_ZSH_USER"
+
+    for sys in $NO_USER_SESSION_SYSTEMS; do
+        if echo "$SPREAD_SYSTEM" | grep "${sys}-"; then
+            exit 0
+        fi
+    done
+
+    touch ./has-sessions
+
+    # setup for systems supporting user sessions
+    for uid in $(id -u test) $(id -u "$TEST_ZSH_USER"); do
```

Comment:
Do we need to wrap $( ) in "$(  )" here?

Reply:
nah, it's a uid, no spaces, just a number

Reply:
If shellcheck does not complain then yes.